tests/test_evaluate_command.py

```diff
@@ -0,0 +1,78 @@
+from unittest.mock import patch
+
+import pytest
+from click.testing import CliRunner
+from yacs.config import CfgNode
+
+from datasetinsights.commands.evaluate import cli
+from datasetinsights.estimators.factory import EstimatorFactory
+
+
+@pytest.mark.parametrize(
+    "args",
+    [
+        [
+            "evaluate",
+            "--config=tests/configs/faster_rcnn_groceries_real_test.yaml",
+            "--checkpoint-file=checkpoint_file.txt",
+            f"--data-root=tests/datasets",
+        ],
+        [
+            "evaluate",
+            "-c",
+            "tests/configs/faster_rcnn_groceries_real_test.yaml",
+            "-d",
+            "tests/datasets",
+            "-p",
+            "--checkpoint=checkpoint_file.txt",
+        ],
+    ],
+)
+@patch("builtins.open")
+@patch.object(CfgNode, "load_cfg")
+@patch.object(EstimatorFactory, "create")
+def test_evaluate_except_called_once(
+    estimator_factory_create_mock, cfg_node_mock, open_mock, args
+):
+    # arrange
+    runner = CliRunner()
+    # act
+
+    runner.invoke(cli, args)
+    # assert
+    open_mock.assert_called_once_with(
+        "tests/configs/faster_rcnn_groceries_real_test.yaml", "r"
+    )
+    cfg_node_mock.assert_called_once()
+    estimator_factory_create_mock.assert_called_once()
+    estimator_factory_create_mock.return_value.evaluate.assert_called_once()
+
+
+@pytest.mark.parametrize(
```

Comment:
Same as above. Why use `@pytest.mark.parametrize` for a single test input value?

Reply:
There are two inputs. It would just be a duplication of the below code. Plus the fact that more can be added this is just the start

Reply:
Why have two inputs that test exactly the same behavior?

Reply:
The purpose of the test is to validate that the config file is not loaded and the estimator factory is not called. 

The purpose of unittests is to validate that for a given input a given output expected. It is possible to have a program with different inputs and the same behaviour. How else would you test that the behavior is the same? 

`download evaluate
`
and 

`download evaluate --config tests/configs/faster_rcnn_groceries_real_test.yaml --data-root invalid-data-root`

From https://en.wikipedia.org/wiki/Unit_testing
Writing and maintaining unit tests can be made faster by using parameterized tests. These allow the execution of one test multiple times with different input sets, thus reducing test code duplication. Unlike traditional unit tests, which are usually closed methods and test invariant conditions, parameterized tests take any set of parameters.

Reply:
From a design perspective it might make more sense to have the cli throw an exception, but this does not change the purpose of using parametrized test.